src/q_0819/tanacchi/mod.rs

```diff
@@ -0,0 +1,108 @@
+use anyhow::{bail, Result};
+use std::io::{Read, Write};
+use std::string::String;
+
+#[allow(dead_code)]
+fn parse(input: &mut std::str::SplitWhitespace) -> Result<i32> {
+    if let Some(s) = input.next() {
+        Ok(s.parse::<i32>()?)
+    } else {
+        eprintln!("Number of input must be more than 1.");
+        bail!("Invalid number of input.")
+    }
+}
+
+#[allow(dead_code)]
+fn main(src: &mut impl Read, dst: &mut impl Write) -> Result<()> {
+    let mut s = String::new();
+    src.read_to_string(&mut s)?;
+
+    let mut input = s.split_whitespace();
+    let x = parse(&mut input)?;
+    let y = parse(&mut input)?;
+
+    if y == 0 {
+        bail!("The second number must not be 0.");
+    }
```

Comment:
[nits]
ここ僕が使うの忘れてたんですがanyhow::ensure使うと簡潔に書けます
```suggestion
    ensure!(y != 0, "The second number must not be 0.");
```

Reply:
おお！すっきりしていいですね！:+1: